website/en/docs/types/empty.md

```diff
@@ -0,0 +1,29 @@
+---
+layout: guide
+---
+
+Empty type has no values, it is also known as [bottom type](https://en.wikipedia.org/wiki/Bottom_type).
+
+The `empty` type is a subtype of _every_ type.
+
+For example, `empty` type is the return type of a function that throws exception:
+
+```js
+// @flow
+function error(message: string): empty {
+  throw new Error(message)
+}
+```
+
+or never returns:
+
+```js
+// @flow
+function loop(): empty {
```

Comment:
This actually errors, but I thought that it shouldn't? https://flow.org/try/#0GYVwdgxgLglg9mABAGznADgCgJQC5ECmAtulAJ6IDeAUAJADuAFjMgZlAE4gHZUC+1PkA

Reply:
Hmmm, this function returns `void`, that's perhaps why?

Reply:
It shouldn't

Reply:
https://github.com/facebook/flow/issues/4088

Reply:
Oh yes you're right, my bad. But I guess that flow cannot understand that we never exit the loop.